### PR TITLE
[comp/otelcol/extension] Fix extension check

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?
This PR addresses 2 issues in the extension check logic:
- It now checks for extensions against the enhanced config, instead of the provided config. In fact, the converter logic can add extensions, so some extensions will be present in the enhanced config but not provided config.
- It checks map `supportedDebugExtensions` against `extension.Type().String()` instead of `extension.String()`. Checking for Type will strip the custom `/<anything>`.

This PR also adds a test that fails without the fix.

### Describe how to test/QA your changes
Run converged agent with pprof, health_check and zpages extensions with a suffix in name (e.g. pprof/custom) and ensure that you see log "Found debug extension at" for all three extensions.

also `curl -k https://localhost:7777` and ensure that there is a sources block, e.g.
```
  "sources": {
    "health_check/dd-autoconfigured": {
      "url": "localhost:13133",
      "crawl": false
    },
    "pprof/dd-autoconfigured": {
      "url": "localhost:1777",
      "crawl": false
    },
    "zpages/dd-autoconfigured": {
      "url": "localhost:55679",
      "crawl": true
    }
  },

```

### CODE REVIEW

File: comp/otelcol/extension/impl/extension.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func (ext *ddExtension) Start(_ context.Context, host component.Host) error {
 
 	extensions := host.GetExtensions()
 	for extension := range extensions {
-		extractor, ok := supportedDebugExtensions[extension.String()]
+		extractor, ok := supportedDebugExtensions[extension.Type().String()]
```
Comment (truthbk) [inline comp/otelcol/extension/impl/extension.go:79]: I assume this `Type()` call strips what follows the `/` in the extension name?

File: comp/otelcol/extension/impl/extension_test.go [hunk 2]
```diff
@@ -86,6 +88,15 @@ func TestExtensionHTTPHandler(t *testing.T) {
 	require.NoError(t, err)
 
 	ddExt := ext.(*ddExtension)
+	ddExt.telemetry.Logger = zap.New(zap.NewNop().Core())
+
+	host := newHostWithExtensions(
```
Comment (truthbk) [inline comp/otelcol/extension/impl/extension_test.go:93]: 🍰

**Review by truthbk — APPROVED** [review]
Looks good; just a small question for my reference. 👍

File: comp/otelcol/extension/impl/extension.go (hunk 1 above)
Comment (mackjmr) [reply comp/otelcol/extension/impl/extension.go:79]: Yes, a component ID comprises of a [Type and a Name](https://github.com/open-telemetry/opentelemetry-collector/blob/component/v0.107.0/component/identifiable.go#L19-L22). The Type is what is before the /, and the name is what is after the /.

**Review by dinooliva — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**